src/vos/vos_io.c

```diff
@@ -733,7 +733,7 @@ akey_update_recx(daos_handle_t toh, daos_epoch_t epoch, uint32_t pm_ver,
 	ent.ei_rect.rc_ex.ex_hi = recx->rx_idx + recx->rx_nr - 1;
 	ent.ei_ver = pm_ver;
 	ent.ei_inob = rsize;
-	if (iod_csum)
+	if (daos_csum_isvalid(iod_csum))
```

Comment:
Should we check whether csum is enabled here and check validity?   It seems like if the container doesn't enable checksums, we should always set this to 0.  If it does have checksums, we should probably return an error

Reply:
Yes, but the ability to check if checksums is enabled for a container doesn't exist yet. I'm waiting on DAOS-2180 which will enable getting container properties on the server side.